---
title: "try-catch-finally (C# Reference)"
ms.custom: na
ms.date: 09/19/2016
ms.devlang: 
  - CSharp
ms.prod: visual-studio-dev14
ms.reviewer: na
ms.suite: na
ms.technology: 
  - devlang-csharp
ms.tgt_pltfrm: na
ms.topic: article
ms.assetid: a1b443b0-ff7a-43ab-b835-0cc9bfbd15ca
caps.latest.revision: 23
translation.priority.ht: 
  - de-de
  - ja-jp
---
# try-catch-finally (C# Reference)
A common usage of `catch` and `finally` together is to obtain and use resources in a `try` block, deal with exceptional circumstances in a `catch` block, and release the resources in the `finally` block.  
  
 For more information and examples on re-throwing exceptions, see [try-catch](../Topic/try-catch%20\(C%23%20Reference\).md) and [Throwing Exceptions](assetId:///72bdd157-caa9-4478-9ee3-cb4500b84528). For more information about the `finally` block, see [try-finally](../vs140/try-finally--C#-Reference-.md).  
  
## Example  
 [!CODE [csrefKeywordsExceptions#1](../CodeSnippet/VS_Snippets_VBCSharp/csrefKeywordsExceptions#1)]  
  
## C# Language Specification  
 [!INCLUDE[CSharplangspec](../vs140/includes/Csharplangspec_md.md)]  
  
## See Also  
 [C# Reference](../vs140/C#-Reference.md)   
 [C# Programming Guide](../vs140/C#-Programming-Guide.md)   
 [C# Keywords](../Topic/C%23%20Keywords.md)   
 [try, throw, and catch Statements (C++)](../vs140/try--throw--and-catch-Statements--C---.md)   
 [Exception Handling Statements](../vs140/Exception-Handling-Statements--C#-Reference-.md)   
 [throw](../Topic/throw%20\(C%23%20Reference\).md)   
 [Throwing Exceptions](assetId:///72bdd157-caa9-4478-9ee3-cb4500b84528)   
 [using Statement (C# Reference)](../Topic/using%20Statement%20\(C%23%20Reference\).md)